### PR TITLE
catch wrong input format

### PR DESCRIPTION
**Pull request by burnout87**
for both catalog and scw_list

### CODE REVIEW

**Comment by burnout87** [comment]
@volodymyrss do we support scw_list in space separated format? Example:

'066500000010.001 066500010010.001 066500020010.001 066500030010.001 066500040010.001'

**Comment by volodymyrss** [comment]
> @volodymyrss do we support scw_list in space separated format? Example:
> 
> '066500000010.001 066500010010.001 066500020010.001 066500030010.001 066500040010.001'

frontend field rejects it, so dispatcher should as well.

**Review by volodymyrss — APPROVED** [review]
seems ok. Thanks!

File: cdci_data_analysis/analysis/instrument.py [hunk 1]
```diff
@@ -539,48 +553,67 @@ def set_input_products(self, par_dic, input_file_path,input_prod_list_name):
...
+                                           '<a href=https://docs.astropy.org/en/stable/api/astropy.io.ascii.Ecsv.html>ECSV</a> table.')
             self.set_par('user_catalog', catalog_object)
-            # TODO set selected_catalog ? in this way it will show up in the email url link
-            # and used consistently for the job_id generation
             self.set_par('selected_catalog', json.dumps(catalog_object.get_dictionary()))
-            # TODO verify that the file path is not needed in the frontend (should not)
+            # TODO not needed in the frontend
             par_dic.pop('user_catalog_file', None)
         else:
+            # TODO is this case still in use?
```
Comment (burnout87) [inline cdci_data_analysis/analysis/instrument.py:586]: Is this still used for catalogs?

File: cdci_data_analysis/analysis/instrument.py [hunk 2]
```diff
@@ -539,48 +553,67 @@ def set_input_products(self, par_dic, input_file_path,input_prod_list_name):
...
+            if type(par_dic['user_catalog_dictionary']) == dict:
+                catalog_to_build = par_dic['user_catalog_dictionary']
+            else:
+                catalog_to_build = json.loads(par_dic['selected_catalog'])
+            if catalog_to_build is not None:
+                try:
+                    catalog_dic = build_catalog(catalog_to_build)
+                except RuntimeError:
+                    raise RequestNotUnderstood("catalog format not valid")
+                self.set_par('user_catalog', catalog_dic)
```
Comment (burnout87) [inline cdci_data_analysis/analysis/instrument.py:615]: and also this?

**Review by burnout87 — COMMENTED** [review]
just one question regarding the setting of a catalog

File: cdci_data_analysis/analysis/instrument.py (hunk 1 above)
Comment (volodymyrss) [reply cdci_data_analysis/analysis/instrument.py:586]: I think that's what passed in some cases when frontend send the table. Well, you can verify if you want.

File: cdci_data_analysis/analysis/instrument.py (hunk 2 above)
Comment (volodymyrss) [reply cdci_data_analysis/analysis/instrument.py:615]: there are several ways of what frontend passes. It's good to verify, but it is not, in principle, up to the user. But ok.

**Review by volodymyrss — CHANGES_REQUESTED** [review]
let's then verify all the ways frontend passes catalog

File: cdci_data_analysis/analysis/instrument.py (hunk 1 above)
Comment (burnout87) [reply cdci_data_analysis/analysis/instrument.py:586]: Just checked, and it is indeed passed from the frontend!

Comment (burnout87) [reply cdci_data_analysis/analysis/instrument.py]: this is tested now

File: cdci_data_analysis/analysis/instrument.py (hunk 2 above)
Comment (burnout87) [reply cdci_data_analysis/analysis/instrument.py]: to me, it's not entirely clear how the `user_catalog_dictionary` should be used and when.

it's not used from the frontend though, so I suppose it's some legacy code

**Review by volodymyrss — DISMISSED** [review]

File: cdci_data_analysis/analysis/instrument.py (hunk 2 above)
Comment (volodymyrss) [reply cdci_data_analysis/analysis/instrument.py]: possibly

Comment (burnout87) [reply cdci_data_analysis/analysis/instrument.py]: Just for the record, I made an advanced search and found out that `user_catalog_dictionary` was used in the past in some unit test, and nowhere else. So, we can get ride of it.

**Review by volodymyrss — APPROVED** [review]

**Pull request merged**